src/test/resources/processor_test_files/2184_CastArithmeticOperand/NOCOMPILE_NullOperandType.java

```diff
@@ -0,0 +1,8 @@
+/*
+When we cannot resolve the type of the left or right hand operands in a binary operator, we get a NullPointerException.
+This test checks that such an exception doesn't happen.
+*/
+class NOCOMPILE_NullOperandType {
+    float twoThirds = 2/3; // Noncompliant; THIS LINE WAS ONLY ADDED TO MAKE THE TEST FIND A VIOLATION IN THIS FILE SO THE NEXT LINE CAN BE ACTUALLY TESTED!
+    long var = System.currentTimeMillis() - message.getJMSTimestamp();
```

Comment:
I suggest adding the "crashy statement" in the opposite configuration as well, so we're confident that the fix works wherever the unknown type appears in the binary operator.

```suggestion
    long rhsUnknown = System.currentTimeMillis() - message.getJMSTimestamp();
    long lhsUnknown = message.getJMSTimestamp() - System.currentTimeMillis();
```

Reply:
Done!